pkg/controller/reconciler/deploymentresource_reconciler.go

```diff
@@ -0,0 +1,374 @@
+	// 	2. If the deletion completed successfully, then remove the `radapp.io/deployment-resource-finalizer` finalizer from the resource and continue processing.
+	// 	3. If the operation failed, then update the `status.phrase` and `status.message` as `Failed`.
+	// 2. If the `DeploymentTemplate` is being deleted, then process deletion:
+	// 	1. Send a DELETE operation to the Radius API to delete the resource specified in the `spec.resourceId` field.
+	// 	2. Continue processing.
+	// 3. If the `DeploymentTemplate` is not being deleted then process this as a create or update:
+	// 	1. Set the `status.phrase` for the `DeploymentResource` to `Ready`.
+	// 	2. Continue processing.
+	//
+	// We do it this way because it guarantees that we only have one operation going at a time.
+
+	if deploymentResource.Status.Operation != nil {
+		result, err := r.reconcileOperation(ctx, &deploymentResource)
+		if err != nil {
+			logger.Error(err, "Unable to reconcile in-progress operation.")
+			return ctrl.Result{}, err
+		} else if result.IsZero() {
+			// NOTE: if reconcileOperation completes successfully, then it will return a "zero" result,
+			// this means the operation has completed and we should continue processing.
+			logger.Info("Operation completed successfully.")
+		} else {
+			logger.Info("Requeueing to continue operation.")
+			return result, nil
+		}
+	}
+
+	if deploymentResource.DeletionTimestamp != nil {
+		return r.reconcileDelete(ctx, &deploymentResource)
+	}
+
+	logger.Info("Resource is in desired state.", "resourceId", deploymentResource.Spec.Id)
+
+	deploymentResource.Status.Phrase = radappiov1alpha3.DeploymentResourcePhraseReady
+	deploymentResource.Status.ProviderConfig = deploymentResource.Spec.ProviderConfig
+	deploymentResource.Status.RootFileName = deploymentResource.Spec.RootFileName
+	deploymentResource.Status.Id = deploymentResource.Spec.Id
+	err = r.Client.Status().Update(ctx, &deploymentResource)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.EventRecorder.Event(&deploymentResource, corev1.EventTypeNormal, "Reconciled", "Successfully reconciled resource.")
+	return ctrl.Result{}, nil
+}
+
+// reconcileOperation reconciles a DeploymentResource that has an operation in progress.
+func (r *DeploymentResourceReconciler) reconcileOperation(ctx context.Context, deploymentResource *radappiov1alpha3.DeploymentResource) (ctrl.Result, error) {
+	logger := ucplog.FromContextOrDiscard(ctx)
+
+	if deploymentResource.Status.Operation.OperationKind == radappiov1alpha3.OperationKindDelete {
+		providerConfig := sdkclients.ProviderConfig{}
+		err := json.Unmarshal([]byte(deploymentResource.Spec.ProviderConfig), &providerConfig)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to unmarshal providerConfig: %w", err)
+		}
+
+		poller, err := r.Radius.Resources(providerConfig.Deployments.Value.Scope, deploymentResourceType).ContinueDeleteOperation(ctx, deploymentResource.Status.Operation.ResumeToken)
```

Comment:
For this step, it looks like we're resuming the operation, and we need the resource ID of the deployment to do that. I think we're already capturing that in `status.id` so we don't also need the provider config?